xrpl-hooks-docs/md/hooks-guard-call-non-const.md

```diff
@@ -0,0 +1,4 @@
+# hooks-guard-call-non-const
```

Comment:
to add a doc page for a new check, it's also necessary to add it to the index at xrpl-hooks-docs/docs.ts (by hand - no code generation here, sorry :-) )